account_invoice_import_invoice2data/wizard/account_invoice_import.py

```diff
@@ -90,9 +90,79 @@ def invoice2data_to_parsed_inv(self, invoice2data_res):
             'date_start': invoice2data_res.get('date_start'),
             'date_end': invoice2data_res.get('date_end'),
             'description': invoice2data_res.get('description'),
+            'mapping_lines': invoice2data_res.get('lines', [])
             }
         if 'amount_untaxed' in invoice2data_res:
             parsed_inv['amount_untaxed'] = invoice2data_res['amount_untaxed']
         if 'amount_tax' in invoice2data_res:
             parsed_inv['amount_tax'] = invoice2data_res['amount_tax']
         return parsed_inv
+
+    @api.model
+    def _prepare_create_invoice_vals(self, parsed_inv):
+        """ Extend invoice with line mapping. """
+        vals = super(AccountInvoiceImport, self)._prepare_create_invoice_vals(
+            parsed_inv)
+        partner = self.env['res.partner'].browse(vals['partner_id'])
+        template = self.env['invoice2data.template'].search([
+            ('related_partner', '=', partner.id)
+        ])
+        config = partner.invoice_import_id
+        if (config.invoice_line_method == 'product_mapping' and
+                parsed_inv.get('mapping_lines')):
+            il_vals = []
+
+            # Get default account_id based on the partner
+            partner_account = template.related_partner.default_account_id
```

Comment:
Suggest to use the `account_id` from the import config instead.

Reply:
with a fallback on the expense account of the config's static_product_id.

I'm now thinking that this feature is completely independent of the invoice2data templates and can should be moved to the core method in account_invoice_import/wizard/account_invoice_import.py. What do you think?